modules/solid_mechanics/src/materials/CombinedPlasticityStressUpdate.C

```diff
@@ -0,0 +1,258 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CombinedPlasticityStressUpdate.h"
+#include "Function.h"
+#include "ElasticityTensorTools.h"
+#include "BackstressRadialReturnStressUpdate.h"
+#include "ElasticityTensorTools.h"
+
+registerMooseObject("SolidMechanicsApp", ADCombinedPlasticityStressUpdate);
+registerMooseObject("SolidMechanicsApp", CombinedPlasticityStressUpdate);
+
+template <bool is_ad>
+InputParameters
+CombinedPlasticityStressUpdateTempl<is_ad>::validParams()
+{
+  InputParameters params = BackstressRadialReturnStressUpdateTempl<is_ad>::validParams();
+  params.addClassDescription("This class uses the discrete material in a radial return isotropic"
+                             "plasticity model.  This class is one of the basic radial return "
+                             "constitutive models, yet it can be used in conjunction with other "
+                             "creep and plasticity materials for more complex simulations.");
+  params.addParam<Real>("Q", 0.0, "Saturation value for isotropic hardening (Q in Voce model)");
+  params.addParam<Real>("b", 0.0, "Rate constant for isotropic hardening (b in Voce model)");
+  params.addParam<FunctionName>("yield_stress_function",
+                                "Yield stress as a function of temperature");
+  params.addParam<Real>("yield_stress", "The point at which plastic strain begins accumulating");
+  params.addParam<FunctionName>("isotropic_hardening_function",
+                                "True stress as a function of plastic strain");
+  params.addParam<Real>("isotropic_hardening_constant", "Isotropic hardening slope");
+  params.addCoupledVar("temperature", 0.0, "Coupled Temperature");
+  params.addDeprecatedParam<std::string>(
+      "plastic_prepend",
+      "",
+      "String that is prepended to the plastic_strain Material Property",
+      "This has been replaced by the 'base_name' parameter");
```

Comment:
You can get rid of this deprecated parameter.  The other class needed it but you won't because you are creating a new class.
```suggestion
```